pkg/apis/metal/v1alpha1/types_worker.go

```diff
@@ -16,5 +15,15 @@ type WorkerStatus struct {
 	// a version that is still in use gets removed from this componentconfig it cannot reconcile anymore existing `Worker`
 	// resources that are still using this version. Hence, it stores the used versions in the provider status to ensure
 	// reconciliation is possible.
-	MachineImages []config.MachineImage `json:"machineImages,omitempty"`
+	MachineImages []MachineImage `json:"machineImages,omitempty"`
```

Comment:
I factored this out into https://github.com/metal-stack/gardener-extension-provider-metal/pull/292.